src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsMatchingPatternsAction.java

```diff
@@ -0,0 +1,147 @@
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(path)) {
+            return !directoryStream.iterator().hasNext();
+        }
+    }
+
+    // if 'file' is on a different node, this FileCallable will be transferred to that node and executed there.
+    public static final class DeleteFileCallable implements FileCallable<Void> {
+        private static final long serialVersionUID = -7712623306942117400L;
+        private final File archiveRootFile;
+
+        public DeleteFileCallable(File archiveRootFile) {
+            this.archiveRootFile = archiveRootFile;
+        }
+
+        @Override 
+        public Void invoke(File file, VirtualChannel channel) throws IOException {
+            deleteFileOrLogError(file);
+            return null;
+        }
+
+        void deleteEmptyDirectoryAndParent(File directory) throws IOException {
+            if (isDirectoryEmpty(directory.toPath()) && hasValidParent(directory.getParentFile())) {
+                Util.deleteFile(directory);
+                deleteParentDirectories(directory.getParentFile());
+            }
+        }
+
+        boolean hasValidParent(File parent) {
+            return this.archiveRootFile.equals(parent);
+        }
+
+        void deleteParentDirectories(File directory) {
+            File parent = directory;
+
+            while (parent != null && !parent.equals(this.archiveRootFile)) {
+                deleteDirectory(parent);
+                parent = parent.getParentFile();
+            }
+        }
+
+        void deleteDirectory(File directory) {
+            boolean deleteSuccess = directory.delete();
+            if(!deleteSuccess) {
+                LOG.log(Level.FINE, "Deletion of directory failed: " + directory.getAbsolutePath());
+            }
+        }
+
+        void deleteFileOrLogError(File file) throws IOException {
+            if (file.isFile()) {
+                LOG.log(Level.FINE, "Deleting " + file.getName());
+                Util.deleteFile(file);
+                deleteEmptyDirectoryAndParent(file.getParentFile());
+
+                // Return early when the condition is met
+                return;
+            }
+            LOG.log(Level.FINE, file + " is neither a directory nor a regular file.");
+        }
+
+        /**
+         * Overrides the 'checkRoles' method to fulfill the requirement of implementing RoleSensitive.
+         *
+         * @param checker The RoleChecker used for access control and security purposes.
+         * @throws SecurityException If there are security-related issues.
+         */
+        @Override
+        public void checkRoles(RoleChecker checker) throws SecurityException {
+            //Nothing is being done here even though a call to RoleChecker#check(…​) is expected.
+        }
+    }
+
+    @Override
+    public void perform(Run<?, ?> run) throws IOException, InterruptedException {
+        VirtualFile vRoot = run.getArtifactManager().root();
```

Comment:
prefer Java naming convention so rather virtualRoot over vRoot but in fact this name reflects to the type while it should describe content or purpose

Reply:
I've updated the variable name to 'virtualRoot' to better reflect its content or purpose. Thank you.